rllib/core/learner/torch/torch_learner.py

```diff
@@ -176,7 +176,27 @@ def compute_gradients(
     def apply_gradients(self, gradients_dict: ParamDict) -> None:
         # Set the gradient of the parameters.
         for pid, grad in gradients_dict.items():
-            self._params[pid].grad = grad
+            # If updates should not be skipped turn `nan` gradients to zero.
```

Comment:
Wait, I'm confused. We have this block here further below, which I think does the exact same thing: skips the entire `optim.step` in case any gradient is non-finite (inf or nan).

```
                # `step` the optimizer (default), but only if all gradients are finite.
                elif all(
                    param.grad is None or torch.isfinite(param.grad).all()
                    for group in optim.param_groups
                    for param in group["params"]
                ):
```

Can you check and see whether these two logics can be consolidated?

Kind of like this:
* If user sets this flag (default=False), the optimizer will skip the update step entirely (+ warning raised by RLlib).
* If user does NOT set this flag (default behavior), grads that are non-finite will be set to 0.0 (+ warning raised by RLlib).